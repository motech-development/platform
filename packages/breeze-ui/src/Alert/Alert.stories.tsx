import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { boolean, select, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import BaseStyles from '../BaseStyles/BaseStyles';
import Alert from './Alert';

const stories = storiesOf('Alert', module);
const colours = {
  Danger: 'danger',
  Primary: 'primary',
  Secondary: 'secondary',
  Success: 'success',
};
const margins = {
  Large: 'lg',
  Medium: 'md',
  Small: 'sm',
};
const message = () => text('Message', 'Hello world');
const colour = () => select('Colour', colours, 'primary');
const spacing = () => select('Spacing', margins, 'md') as 'sm' | 'md' | 'lg';
const dismissable = () => boolean('Dismissable', false);

stories.addDecorator(withKnobs);

stories.add('Various alerts', () => (
  <>
    <BaseStyles />

    <Alert
      message={message()}
      colour={colour()}
      dismissable={dismissable()}
      spacing={spacing()}
    />

    <Alert
      message={message()}
      colour={colour()}
      dismissable={dismissable()}
      icon={() => <FontAwesomeIcon icon={faExclamationTriangle} />}
      spacing={spacing()}
    />
  </>
));
