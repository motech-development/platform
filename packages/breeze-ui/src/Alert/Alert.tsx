import { faTimesCircle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { ElementType, FC, memo, useState } from 'react';
import styled, { ThemeProvider } from 'styled-components';

type AlertSpacing = 'sm' | 'md' | 'lg';

interface IAlertTheme {
  [name: string]: {
    background: string;
    colour: string;
  };
}

const alertTheme: IAlertTheme = {
  danger: {
    background: 'rgb(199,56,79)',
    colour: '#fff',
  },
  primary: {
    background: '#2e9dc8',
    colour: '#fff',
  },
  secondary: {
    background: '#f6f9fc',
    colour: '#333',
  },
  success: {
    background: 'rgb(0,128,93)',
    colour: '#fff',
  },
};

interface IBaseAlert {
  colour: keyof typeof alertTheme;
  spacing: AlertSpacing;
}

const BaseAlert = styled.div<IBaseAlert>`
  ${({ colour, spacing, theme }) => `
    background-color: ${theme[colour].background};
    color: ${theme[colour].colour};
    margin-bottom: ${(() => {
      switch (spacing) {
        case 'lg':
          return '20px';
        case 'md':
          return '10px';
        default:
          return '5px';
      }
    })()};
    padding: 12px 56px 12px 20px;
    position: relative;
  `}
`;

const AlertIconWrapper = styled.div`
  display: inline-block;
  margin-right: 20px;
`;

interface IAlertDismissButton {
  colour: keyof typeof alertTheme;
}

const AlertDismissButton = styled.button<IAlertDismissButton>`
  ${({ colour, theme }) => `
    appearance: none;
    background-color: ${theme[colour].background};
    border: 0;
    color: ${theme[colour].colour};
    cursor: pointer;
    font-size: inherit;
    margin: 0;
    padding: 0;
    position: absolute;
    right: 20px;
    top: 12px;
  `}
`;

export interface IAlertProps {
  colour?: keyof typeof alertTheme;
  dismissable?: boolean;
  icon?: ElementType;
  message: string;
  spacing?: AlertSpacing;
}

const Alert: FC<IAlertProps> = ({
  colour = 'primary',
  dismissable = false,
  icon: Icon = null,
  message,
  spacing = 'md',
}) => {
  const [visible, setVisiblity] = useState(true);
  const dismiss = () => {
    setVisiblity(false);
  };

  return (
    <ThemeProvider theme={alertTheme}>
      {visible && (
        <BaseAlert role="alert" colour={colour} spacing={spacing}>
          {Icon && (
            <AlertIconWrapper>
              <Icon />
            </AlertIconWrapper>
          )}

          {message}

          {dismissable && (
            <AlertDismissButton
              type="button"
              aria-label="Dismiss"
              colour={colour}
              onClick={dismiss}
            >
              <FontAwesomeIcon icon={faTimesCircle} />
            </AlertDismissButton>
          )}
        </BaseAlert>
      )}
    </ThemeProvider>
  );
};

export default memo(Alert);
