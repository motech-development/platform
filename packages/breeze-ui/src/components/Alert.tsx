import { Transition } from '@headlessui/react';
import { XIcon } from '@heroicons/react/solid';
import { FC, Fragment, ReactNode, useEffect, useState } from 'react';
import { className, themeClass, spacingClass } from '../utils/className';
import TSpacing from '../utils/spacing';
import TTheme from '../utils/theme';

export interface IAlertProps {
  colour?: TTheme;
  dismissable?: boolean | number;
  dismissText?: string;
  icon?: ReactNode;
  message: string;
  spacing?: TSpacing;
  onDismiss?: () => void;
}

const Alert: FC<IAlertProps> = ({
  colour = 'primary',
  dismissable = false,
  dismissText = 'Dismiss',
  icon,
  message,
  spacing = 'md',
  onDismiss,
}) => {
  const [visible, setVisiblity] = useState(true);
  const dismiss = () => {
    setVisiblity(false);

    if (onDismiss) {
      onDismiss();
    }
  };

  useEffect(
    () => () => {
      setVisiblity(false);
    },
    [],
  );

  useEffect(() => {
    let timeout: ReturnType<typeof setTimeout>;

    if (typeof dismissable === 'number') {
      timeout = setTimeout(dismiss, dismissable);
    }

    return () => {
      if (timeout) {
        clearTimeout(timeout);
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Transition
      show={visible}
      as={Fragment}
      enter="transition ease-in duration-100"
      enterFrom="opacity-0"
      enterTo="opacity-100"
      leave="transition ease-in duration-100"
      leaveFrom="opacity-100"
      leaveTo="opacity-0"
    >
      {/* @tailwind: bg-blue-50 bg-gray-50 bg-green-50 bg-red-50 bg-yellow-50 */}
      {/* @tailwind: mb-0 mb-2 mb-4 mb-6 */}
      <div
        className={className(
          'p-4',
          themeClass(colour, 'bg-{theme}-50'),
          spacingClass(spacing, 'mb-{spacing}'),
        )}
      >
        <div className="flex">
          <div className="flex-shrink-0">
            {/* @tailwind: text-blue-400 text-gray-400 text-green-400 text-red-400 text-yellow-400 */}
            <div
              className={className(
                'h-5 w-5',
                themeClass(colour, 'text-{theme}-400'),
              )}
            >
              {icon}
            </div>
          </div>
          <div className="ml-3">
            {/* @tailwind: text-blue-800 text-gray-800 text-green-800 text-red-800 text-yellow-800  */}
            <p
              className={className(
                'text-sm font-medium',
                themeClass(colour, 'text-{theme}-800'),
              )}
            >
              {message}
            </p>
          </div>
          <div className="ml-auto pl-3">
            <div className="-mx-1.5 -my-1.5">
              {/* @tailwind: bg-blue-50 text-blue-500 hover:bg-blue-100 focus:ring-offset-blue-50 focus:ring-blue-600 */}
              {/* @tailwind: bg-gray-50 text-gray-500 hover:bg-gray-100 focus:ring-offset-gray-50 focus:ring-gray-600 */}
              {/* @tailwind: bg-green-50 text-green-500 hover:bg-green-100 focus:ring-offset-green-50 focus:ring-green-600 */}
              {/* @tailwind: bg-red-50 text-red-500 hover:bg-red-100 focus:ring-offset-red-50 focus:ring-red-600 */}
              {/* @tailwind: bg-yellow-50 text-yellow-500 hover:bg-yellow-100 focus:ring-offset-yellow-50 focus:ring-yellow-600 */}
              {dismissable && (
                <button
                  className={className(
                    'inline-flex p-1.5 focus:outline-none focus:ring-2 focus:ring-offset-2',
                    themeClass(colour, 'bg-{theme}-50'),
                    themeClass(colour, 'text-{theme}-500'),
                    themeClass(colour, 'hover:bg-{theme}-100'),
                    themeClass(colour, 'focus:ring-offset-{theme}-50'),
                    themeClass(colour, 'focus:ring-{theme}-600'),
                  )}
                  type="button"
                  onClick={dismiss}
                >
                  <span className="sr-only">{dismissText}</span>

                  <XIcon className="h-5 w-5" aria-hidden="true" />
                </button>
              )}
            </div>
          </div>
        </div>
      </div>
    </Transition>
  );
};

export default Alert;
