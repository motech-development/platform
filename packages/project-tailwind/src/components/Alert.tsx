import { Transition } from '@headlessui/react';
import { XMarkIcon } from '@heroicons/react/24/solid';
import { Fragment, ReactNode, useCallback, useEffect, useState } from 'react';
import {
  Sizing,
  Themes,
  TSizing,
  TTheme,
  useTailwind,
} from '../utilities/tailwind';

/** Alert component properties */
export interface IAlertProps {
  /** Sets whether alert can be dismissed or a timer, in milliseconds, to automatically dismiss */
  dismissable?: boolean | number;

  /** Dismiss button text  */
  dismissText?: string;

  /** An icon to display before the alert message */
  icon?: ReactNode;

  /** Alert message text */
  message: string;

  /** Component spacing */
  spacing?: TSizing;

  /** Component theme */
  theme?: TTheme;

  /** Dismiss callback */
  onDismiss?: () => void;
}

/** Default dismiss text label */
const DISMISS_TEXT = 'Dismiss';

/**
 * Display application messages and alerts
 *
 * @param props - Component props
 *
 * @returns Alert component
 */
export function Alert(props: IAlertProps) {
  const {
    dismissable = false,
    dismissText = DISMISS_TEXT,
    icon,
    message,
    spacing = Sizing.NONE,
    theme = Themes.PRIMARY,
    onDismiss,
  } = props;

  const [visible, setVisiblity] = useState(true);

  const { createStyles } = useTailwind(theme, spacing);

  const alertStyles = createStyles({
    classNames: ['p-4'],
    sizing: {
      lg: ['mb-6'],
      md: ['mb-4'],
      none: ['mb-0'],
      sm: ['mb-2'],
    },
    theme: {
      danger: ['bg-red-50 text-red-800'],
      primary: ['bg-blue-50 text-blue-800'],
      secondary: ['bg-gray-50 text-gray-800'],
      success: ['bg-green-50 text-green-800'],
      warning: ['bg-yellow-50 text-yellow-800'],
    },
  });

  const buttonStyles = createStyles({
    classNames: [
      'inline-flex p-1.5 focus:outline-none focus:ring-2 focus:ring-offset-2',
    ],
    theme: {
      danger: [
        'bg-red-50 text-red-500 hover:bg-red-100 focus:ring-offset-red-50 focus:ring-red-600',
      ],
      primary: [
        'bg-blue-50 text-blue-500 hover:bg-blue-100 focus:ring-offset-blue-50 focus:ring-blue-600',
      ],
      secondary: [
        'bg-gray-50 text-gray-500 hover:bg-gray-100 focus:ring-offset-gray-50 focus:ring-gray-600',
      ],
      success: [
        'bg-green-50 text-green-500 hover:bg-green-100 focus:ring-offset-green-50 focus:ring-green-600',
      ],
      warning: [
        'bg-yellow-50 text-yellow-500 hover:bg-yellow-100 focus:ring-offset-yellow-50 focus:ring-yellow-600',
      ],
    },
  });

  const iconStyles = createStyles({
    classNames: ['h-5 w-5'],
    theme: {
      danger: ['text-red-400'],
      primary: ['text-blue-400'],
      secondary: ['text-gray-400'],
      success: ['text-green-400'],
      warning: ['text-yellow-400'],
    },
  });

  const messageStyle = createStyles({
    classNames: ['text-sm font-medium'],
  });

  const dismiss = useCallback(() => {
    setVisiblity(false);

    if (onDismiss) {
      onDismiss();
    }
  }, [onDismiss]);

  const timeoutCallback = useCallback(() => {
    let timeout: ReturnType<typeof setTimeout> | null = null;

    if (typeof dismissable === 'number') {
      timeout = setTimeout(dismiss, dismissable);
    }

    return timeout;
  }, [dismiss, dismissable]);

  useEffect(() => {
    const timeout = timeoutCallback();

    return () => {
      if (timeout) {
        clearTimeout(timeout);
      }
    };
  }, [timeoutCallback]);

  return (
    <Transition
      show={visible}
      as={Fragment}
      enter="transition ease-in duration-100"
      enterFrom="opacity-0"
      enterTo="opacity-100"
      leave="transition ease-in duration-100"
      leaveFrom="opacity-100"
      leaveTo="opacity-0"
    >
      <div className={alertStyles} role="alert">
        <div className="flex">
          <div className="flex-shrink-0">
            <p className={iconStyles}>{icon}</p>
          </div>

          <div className="ml-3">
            <p className={messageStyle}>{message}</p>
          </div>

          <div className="ml-auto pl-3">
            <div className="-mx-1.5 -my-1.5">
              {dismissable && (
                <button
                  className={buttonStyles}
                  type="button"
                  onClick={dismiss}
                >
                  <span className="sr-only">{dismissText}</span>

                  <XMarkIcon className="h-5 w-5" />
                </button>
              )}
            </div>
          </div>
        </div>
      </div>
    </Transition>
  );
}
