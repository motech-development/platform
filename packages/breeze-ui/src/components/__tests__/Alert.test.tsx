import { act, fireEvent, render } from '@testing-library/react';
import Alert from '../Alert';

describe('Alert', () => {
  let onDismiss: jest.Mock;

  beforeEach(() => {
    onDismiss = jest.fn();
  });

  describe('spacing', () => {
    it('should have the correct spacing by default', () => {
      const { container } = render(<Alert message="Hello world" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have no spacing', () => {
      const { container } = render(
        <Alert message="Hello world" spacing="none" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct small spacing', () => {
      const { container } = render(
        <Alert message="Hello world" spacing="sm" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct medium spacing', () => {
      const { container } = render(
        <Alert message="Hello world" spacing="md" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct large spacing', () => {
      const { container } = render(
        <Alert message="Hello world" spacing="lg" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('theming', () => {
    it('should have the correct colours by default', () => {
      const { container } = render(<Alert message="Hello world" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct colours by for primary', () => {
      const { container } = render(
        <Alert message="Hello world" colour="primary" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct colours by for secondary', () => {
      const { container } = render(
        <Alert message="Hello world" colour="secondary" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct colours by for danger', () => {
      const { container } = render(
        <Alert message="Hello world" colour="danger" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct colours by for success', () => {
      const { container } = render(
        <Alert message="Hello world" colour="success" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should have the correct colours by for warning', () => {
      const { container } = render(
        <Alert message="Hello world" colour="warning" />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should display the correct message', async () => {
    const { findByText } = render(<Alert message="Hello world" />);

    await expect(findByText('Hello world')).resolves.toBeDefined();
  });

  it('should display the an icon if one is set', async () => {
    const { findByTestId } = render(
      <Alert message="Hello world" icon={<div data-testid="icon" />} />,
    );

    await expect(findByTestId('icon')).resolves.toBeDefined();
  });

  it('should display the dismiss button if dismissable', async () => {
    const { findByTestId } = render(
      <Alert dismissable message="Hello world" />,
    );

    await expect(findByTestId('alert-dismiss-text')).resolves.toHaveTextContent(
      'Dismiss',
    );
  });

  it('should display the correct dismissable text', async () => {
    const { findByTestId } = render(
      <Alert dismissable dismissText="Close me" message="Hello world" />,
    );

    await expect(findByTestId('alert-dismiss-text')).resolves.toHaveTextContent(
      'Close me',
    );
  });

  it('should call onDismiss when one is set', async () => {
    const { findByTestId } = render(
      <Alert dismissable message="Hello world" onDismiss={onDismiss} />,
    );
    const dismiss = await findByTestId('alert-dismiss-text');

    fireEvent.click(dismiss);

    expect(onDismiss).toHaveBeenCalled();
  });

  it('should auto dismiss if a time is set', async () => {
    jest.useFakeTimers();

    const { queryByRole } = render(
      <Alert dismissable={5000} message="Hello world" />,
    );

    act(() => {
      jest.runOnlyPendingTimers();
    });

    expect(queryByRole('alert')).not.toBeInTheDocument();
  });

  it('should not be displayed when dismissed', async () => {
    const { findByTestId, queryByRole } = render(
      <Alert dismissable message="Hello world" />,
    );
    const dismiss = await findByTestId('alert-dismiss-text');

    fireEvent.click(dismiss);

    expect(queryByRole('alert')).not.toBeInTheDocument();
  });
});
