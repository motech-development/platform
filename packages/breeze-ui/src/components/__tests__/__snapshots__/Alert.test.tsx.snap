// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert spacing should have no spacing 1`] = `
<div
  class="p-4 bg-blue-50 mb-0"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert spacing should have the correct large spacing 1`] = `
<div
  class="p-4 bg-blue-50 mb-6"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert spacing should have the correct medium spacing 1`] = `
<div
  class="p-4 bg-blue-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert spacing should have the correct small spacing 1`] = `
<div
  class="p-4 bg-blue-50 mb-2"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert spacing should have the correct spacing by default 1`] = `
<div
  class="p-4 bg-blue-50 mb-4 transition ease-in duration-100 opacity-0"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert theming should have the correct colours by default 1`] = `
<div
  class="p-4 bg-blue-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert theming should have the correct colours by for danger 1`] = `
<div
  class="p-4 bg-red-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-red-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-red-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert theming should have the correct colours by for primary 1`] = `
<div
  class="p-4 bg-blue-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-blue-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-blue-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert theming should have the correct colours by for secondary 1`] = `
<div
  class="p-4 bg-gray-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-gray-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-gray-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert theming should have the correct colours by for success 1`] = `
<div
  class="p-4 bg-green-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-green-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-green-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;

exports[`Alert theming should have the correct colours by for warning 1`] = `
<div
  class="p-4 bg-yellow-50 mb-4"
>
  <div
    class="flex"
  >
    <div
      class="flex-shrink-0"
    >
      <div
        class="h-5 w-5 text-yellow-400"
      />
    </div>
    <div
      class="ml-3"
    >
      <p
        class="text-sm font-medium text-yellow-800"
      >
        Hello world
      </p>
    </div>
    <div
      class="ml-auto pl-3"
    >
      <div
        class="-mx-1.5 -my-1.5"
      />
    </div>
  </div>
</div>
`;
